--- ../node/test/simple/test-http-end-throw-socket-handling.js	2013-09-23 13:54:04.000000000 +0200
+++ test/simple/test-http-end-throw-socket-handling.js	2013-10-24 13:52:31.000000000 +0200
@@ -50,10 +50,11 @@
   }
 });
 
+// increase timeout
 setTimeout(function() {
   process.removeListener('uncaughtException', catcher);
   throw new Error('Taking too long!');
-}, 1000).unref();
+},10000).unref();
 
 process.on('uncaughtException', catcher);
 var errors = 0;
