--- ../node-v10/test/simple/test-http-head-request.js	2013-04-30 22:00:43.501305626 -0700
+++ node/test/simple/test-http-head-request.js	2013-05-23 14:56:14.641294803 -0700
@@ -31,7 +31,6 @@
   common.error('req: ' + req.method);
   res.writeHead(200, {'Content-Length': body.length});
   res.end();
-  server.close();
 });
 
 var gotEnd = false;
@@ -48,6 +47,7 @@
       gotEnd = true;
     });
     response.resume();
+    server.close();
   });
   request.end();
 });
