--- ../node/lib/zlib.js	2013-01-11 14:56:45.000000000 +0100
+++ src/main/js/lib/zlib.js	2013-04-11 14:29:53.000000000 +0200
@@ -21,7 +21,7 @@
 
 var binding = process.binding('zlib');
 var util = require('util');
-var Stream = require('stream');
+var Stream = require('stream').Stream;
 var assert = require('assert').ok;
 
 // zlib doesn't provide these, so kludge them in following the same
@@ -304,7 +304,8 @@
   if (this._hadError) return true;
 
   if (this._ended) {
-    return this.emit('error', new Error('Cannot write after end'));
+     return;
+    //return this.emit('error', new Error('Cannot write after end'));
   }
 
   if (arguments.length === 1 && typeof chunk === 'function') {
@@ -332,6 +333,7 @@
 };
 
 Zlib.prototype.reset = function reset() {
+  this._flush = binding.Z_NO_FLUSH; 
   return this._binding.reset();
 };
 
