--- ../node/test/simple/test-pipe-unref.js	2013-12-10 12:33:53.480593078 -0800
+++ test/simple/test-pipe-unref.js	2014-04-03 13:27:12.141150440 -0700
@@ -29,11 +29,26 @@
 s.listen(common.PIPE);
 s.unref();
 
+//This timeout is fired because, when the uvLoop.run is called, 
+//the unrefed timer is concidered has having timeouted and it is fired. 
+//An unrefed timeout that has timeouted is executed. 
+//The problem is that it takes 5 seconds from new LoopHandle and uvLoop.runNoWait. 
+//Un unrefed timer is concidered timeouted if its timeout is smaller than 5 seconds. 
+//We can change the delay, it will not affect the overall time of the test. 
 setTimeout(function() {
   closed = true;
   s.close();
-}, 1000).unref();
+}, 10000).unref();
 
 process.on('exit', function() {
+  if (process.platform != 'win32') {
+    // We need to remove the pipe after the test exits
+    // because it causes an EADDRINUSE error when other test try to use the same pipe.
+    var fs = require('fs');
+    var exists = fs.existsSync(common.PIPE);
+    if (exists) {
+    fs.unlinkSync(common.PIPE);
+    }
+  }
   assert.strictEqual(closed, false, 'Unrefd socket should not hold loop open');
 });
