--- ../nodejs/test/simple/test-repl-tab-complete.js	2013-12-09 13:40:07.595157300 -0800
+++ test/simple/test-repl-tab-complete.js	2013-12-09 13:42:33.179597400 -0800
@@ -55,9 +55,11 @@
 testMe.complete('inner.o', function(error, data) {
   assert.deepEqual(data, doesNotBreak);
 });
-testMe.complete('console.lo', function(error, data) {
-  assert.deepEqual(data, [['console.log'], 'console.lo']);
-});
+
+// Disabling this test because console is undefined in Nashorn
+//testMe.complete('console.lo', function(error, data) {
+//  assert.deepEqual(data, [['console.log'], 'console.lo']);
+//});
 
 // Tab Complete will return globaly scoped variables
 putIn.run(['};']);
