--- ../nodejs/test/simple/test-pipe-unref.js	2013-12-17 11:28:18.690118948 -0800
+++ test/simple/test-pipe-unref.js	2013-12-17 11:51:42.109888011 -0800
@@ -29,11 +29,24 @@
 s.listen(common.PIPE);
 s.unref();
 
+//This timeout is fired because, when the uvLoop.run is called, 
+//the unrefed timer is concidered has having timeouted and it is fired. 
+//An unrefed timeout that has timeouted is executed. 
+//The problem is that it takes 5 seconds from new LoopHandle and uvLoop.runNoWait. 
+//Un unrefed timer is concidered timeouted if its timeout is smaller than 5 seconds. 
+//We can change the delay, it will not affect the overall time of the test. 
 setTimeout(function() {
   closed = true;
   s.close();
-}, 1000).unref();
+}, 7000).unref();
 
 process.on('exit', function() {
+  // We need to remove the pipe after the test exits
+  // because it causes an EADDRINUSE error when other test try to use the same pipe.
+  var fs = require('fs');
+  var exists = fs.existsSync(common.PIPE);
+  if (exists) {
+    fs.unlinkSync(common.PIPE);
+  }
   assert.strictEqual(closed, false, 'Unrefd socket should not hold loop open');
 });
