--- ../node/test/simple/test-module-loading.js	2013-05-05 20:53:41.000000000 +0200
+++ node/test/simple/test-module-loading.js	2013-05-28 19:26:46.000000000 +0200
@@ -175,8 +175,10 @@
 try {
   require(loadOrder + 'file3');
 } catch (e) {
+  // dlopen not supported
+  assert.ok(e);
   // Not a real .node module, but we know we require'd the right thing.
-  assert.ok(e.message.replace(/\\/g, '/').match(/file3\.node/));
+  //assert.ok(e.message.replace(/\\/g, '/').match(/file3\.node/));
 }
 assert.equal(require(loadOrder + 'file4').file4, 'file4.reg', msg);
 assert.equal(require(loadOrder + 'file5').file5, 'file5.reg2', msg);
@@ -184,7 +186,9 @@
 try {
   require(loadOrder + 'file7');
 } catch (e) {
-  assert.ok(e.message.replace(/\\/g, '/').match(/file7\/index\.node/));
+  // dlopen not supported
+  assert.ok(e);
+  //assert.ok(e.message.replace(/\\/g, '/').match(/file7\/index\.node/));
 }
 assert.equal(require(loadOrder + 'file8').file8, 'file8/index.reg', msg);
 assert.equal(require(loadOrder + 'file9').file9, 'file9/index.reg2', msg);
