--- ../nodejs/test/simple/test-stdout-close-catch.js	2014-01-07 15:32:53.556005400 -0800
+++ test/simple/test-stdout-close-catch.js	2014-01-07 16:20:47.154233600 -0800
@@ -28,14 +28,15 @@
 
 var testScript = path.join(common.fixturesDir, 'catch-stdout-error.js');
 
-var cmd = JSON.stringify(process.execPath) + ' ' +
+var cmd = process.execPath + ' ' +
           JSON.stringify(testScript) + ' | ' +
-          JSON.stringify(process.execPath) + ' ' +
+          process.execPath + ' ' +
           '-pe "process.exit(1);"';
 
 var child = child_process.exec(cmd);
 var output = '';
-var outputExpect = { 'code': 'EPIPE',
+var outputExpect = { 'message' : 'write EPIPE',
+                     'code': 'EPIPE',
                      'errno': 'EPIPE',
                      'syscall': 'write' };
 
