--- ../node/test/simple/test-stream2-readable-empty-buffer-no-eof.js	2013-12-09 10:27:27.000000000 +0100
+++ test/simple/test-stream2-readable-empty-buffer-no-eof.js	2014-01-02 11:11:36.000000000 +0100
@@ -58,7 +58,9 @@
           return r.push(new Buffer(0));
         });
       case 4:
-        setTimeout(r.read.bind(r, 0), 10);
+          // Avatar.js, a longer timeout before to read that a Buffer(0) has been 
+          // pushed
+        setTimeout(r.read.bind(r, 0), 100);
         return setTimeout(function() {
           return r.push(new Buffer(0));
         });
