--- ../node/test/simple/test-readline-interface.js	2013-12-06 15:13:28.003097729 -0800
+++ test/simple/test-readline-interface.js	2013-12-06 15:25:17.018281776 -0800
@@ -170,6 +170,7 @@
 
   // sending a multi-byte utf8 char over multiple writes
   var buf = Buffer('☮', 'utf8');
+  var arr = [].slice.call(buf, 0);
   fi = new FakeInput();
   rli = new readline.Interface({ input: fi, output: fi, terminal: terminal });
   callCount = 0;
@@ -177,7 +178,7 @@
     callCount++;
     assert.equal(line, buf.toString('utf8'));
   });
-  [].forEach.call(buf, function(i) {
+  [].forEach.call(arr, function(i) {
     fi.emit('data', Buffer([i]));
   });
   assert.equal(callCount, 0);
