--- ../node/test/simple/test-cluster-bind-twice-v2.js	2013-12-09 10:27:58.000000000 +0100
+++ ./test/simple/test-cluster-bind-twice-v2.js	2014-01-13 17:24:40.000000000 +0100
@@ -117,6 +117,8 @@
 
 function startWorker() {
   var worker = cluster.fork();
+  // Avatar.js specific
+  process.signals.start('SIGTERM');
   worker.on('message', process.send.bind(process));
   process.on('message', worker.send.bind(worker));
   process.on('SIGTERM', function() {
