--- ../node/test/simple/test-tls-getcipher.js	2013-05-21 14:51:32.000000000 +0200
+++ node/test/crypto/test-tls-getcipher.js	2013-05-22 18:42:10.000000000 +0200
@@ -49,7 +49,9 @@
     rejectUnauthorized: false
   }, function() {
     var cipher = client.getCipher();
-    assert.equal(cipher.name, cipher_list[0]);
+    // Was cipher_list[0], but Java automatically picks the safest
+    // that is AES256-SHA
+    assert.equal(cipher.name, cipher_list[1]);
     assert(cipher_version_pattern.test(cipher.version));
     client.end();
     server.close();
