--- ../node/test/simple/test-crypto-stream.js	2013-12-18 10:39:37.000000000 +0100
+++ test/crypto/test-crypto-stream.js	2013-12-18 10:39:08.000000000 +0100
@@ -71,7 +71,7 @@
 cipher.pipe(decipher)
   .on('error', common.mustCall(function end(err) {
     // TypeError: error:06065064:digital envelope routines:EVP_DecryptFinal_ex:bad decrypt
-    assert(/:06065064:/.test(err));
+    assert(/Given final block not properly padded/.test(err));
   }));
 
 cipher.end('Papaya!');  // Should not cause an unhandled exception.
