--- ../node/test/common.js	2014-03-12 11:18:16 -0700
+++ test/common.js	2014-03-14 22:23:43 -0700
@@ -60,7 +60,7 @@
 exports.ddCommand = function(filename, kilobytes) {
   if (process.platform === 'win32') {
     var p = path.resolve(exports.fixturesDir, 'create-file.js');
-    return '"' + process.argv[0] + '" "' + p + '" "' +
+    return process.argv[0] + ' "' + p + '" "' +
            filename + '" ' + (kilobytes * 1024);
   } else {
     return 'dd if=/dev/zero of="' + filename + '" bs=1024 count=' + kilobytes;
@@ -125,6 +125,10 @@
   knownGlobals.push(COUNTER_HTTP_CLIENT_RESPONSE);
 }
 
+// ArrayBuffer is not fully implemented in nashorn (as of jdk8)
+// https://bugs.openjdk.java.net/browse/JDK-8015958
+// https://bugs.openjdk.java.net/browse/JDK-8036743
+global.ArrayBuffer = undefined;
 if (global.ArrayBuffer) {
   knownGlobals.push(ArrayBuffer);
   knownGlobals.push(Int8Array);
@@ -219,3 +223,12 @@
   assert.strictEqual(ret.status, 0);
   assert.strictEqual(ret.error, undefined);
 };
+
+exports.isIPv6Enabled = true;
+var dgram = require('dgram');
+var socket = dgram.createSocket('udp6');
+try {
+    socket.send6('', common.PORT, '::1');
+} catch(error) {
+    exports.isIPv6Enabled = false;
+}
