--- ../node/lib/fs.js	2013-12-16 14:49:26.000000000 +0100
+++ ./src/main/js/lib/fs.js	2014-01-14 10:38:31.000000000 +0100
@@ -62,6 +62,8 @@
   var callback;
   if (DEBUG) {
     var backtrace = new Error;
+    // Nashorn: https://bugs.openjdk.java.net/browse/JDK-8029967
+    Error.captureStackTrace(backtrace);
     callback = debugCallback;
   } else
     callback = missingCallback;
@@ -104,7 +106,7 @@
   }
 
   return function() {
-    return cb.apply(null, arguments);
+    return cb.apply(global, arguments); // test-fs-stat.js
   };
 }
 
