--- ../nodejs/lib/repl.js	2013-12-06 14:29:56.315446400 -0800
+++ src/main/js/lib/repl.js	2013-12-06 14:41:35.670447200 -0800
@@ -535,7 +535,7 @@
         // Get global vars synchronously
         if (this.useGlobal ||
             this.context.constructor &&
-            this.context.constructor.name === 'Context') {
+            this.context.constructor.name === 'Object') {
           var contextProto = this.context;
           while (contextProto = Object.getPrototypeOf(contextProto)) {
             completionGroups.push(Object.getOwnPropertyNames(contextProto));
