--- ../nodejs/lib/repl.js	2013-12-09 18:56:15.294402800 -0800
+++ src/main/js/lib/repl.js	2013-12-10 11:23:45.938789000 -0800
@@ -535,7 +535,7 @@
         // Get global vars synchronously
         if (this.useGlobal ||
             this.context.constructor &&
-            this.context.constructor.name === 'Context') {
+            this.context.constructor.name === 'Object') {
           var contextProto = this.context;
           while (contextProto = Object.getPrototypeOf(contextProto)) {
             completionGroups.push(Object.getOwnPropertyNames(contextProto));
@@ -923,5 +923,5 @@
       // "strict mode" syntax errors
       !e.match(/^SyntaxError: .*strict mode.*/i) &&
       // JSON.parse() error
-      !e.match(/\n {4}at Object.parse \(native\)\n/);
+      !e.match(/^SyntaxError: Invalid JSON/);
 }
