--- ../nodejs/lib/repl.js	2013-12-05 16:43:37.738090700 -0800
+++ src/main/js/lib/repl.js	2013-12-05 16:44:53.691533100 -0800
@@ -535,7 +535,7 @@
         // Get global vars synchronously
         if (this.useGlobal ||
             this.context.constructor &&
-            this.context.constructor.name === 'Context') {
+            this.context.constructor.name === 'Object') {
           var contextProto = this.context;
           while (contextProto = Object.getPrototypeOf(contextProto)) {
             completionGroups.push(Object.getOwnPropertyNames(contextProto));
@@ -918,10 +918,10 @@
   e = e && (e.stack || e.toString());
   return e && e.match(/^SyntaxError/) &&
       // RegExp syntax error
-      !e.match(/^SyntaxError: Invalid regular expression/) &&
+      !e.match(/^SyntaxError: repl/) &&
       !e.match(/^SyntaxError: Invalid flags supplied to RegExp constructor/) &&
       // "strict mode" syntax errors
       !e.match(/^SyntaxError: .*strict mode.*/i) &&
       // JSON.parse() error
-      !e.match(/\n {4}at Object.parse \(native\)\n/);
+      !e.match(/^SyntaxError: Invalid /);
 }
