--- ../node/lib/timers.js	2013-05-05 20:53:40.000000000 +0200
+++ src/main/js/lib/timers.js	2013-05-13 15:13:01.000000000 +0200
@@ -111,7 +111,7 @@
         if (domain)
           domain.exit();
         threw = false;
-      } finally {
+      } catch(ex) { // nashorn workaround https://jbs.oracle.com/bugs/browse/JDK-8013919
         if (threw) {
           process.nextTick(function() {
             list.ontimeout();
