--- ../node/lib/util.js	2014-03-12 11:18:16 -0700
+++ src/main/js/lib/util.js	2014-03-13 13:07:27 -0700
@@ -231,6 +231,14 @@
     return primitive;
   }
 
+  // Java thing, can be a Dynalink function, a java instance, a java class
+  if (typeof(value.__proto__) === 'undefined' &&
+     (typeof(value) == 'object' || typeof(value) == 'function') &&
+     !(value instanceof Object)) { // java
+    // equivalent to toString whatever toString exists or not
+    return "" + value;
+  }
+
   // Look up the keys of the object.
   var keys = Object.keys(value);
   var visibleKeys = arrayToHash(keys);
@@ -261,6 +269,9 @@
     });
   }
 
+  if (!ctx.showHidden && isError(value)) {
+      return formatError(value);
+  }
   // Some type of object without properties can be shortcutted.
   if (keys.length === 0) {
     if (isFunction(value)) {
@@ -439,7 +450,7 @@
       str = ctx.stylize('[Setter]', 'special');
     }
   }
-  if (!hasOwnProperty(visibleKeys, key)) {
+  if (isError(value) || !hasOwnProperty(visibleKeys, key)) {
     name = '[' + key + ']';
   }
   if (!str) {
@@ -582,7 +593,7 @@
 exports.isPrimitive = isPrimitive;
 
 function isBuffer(arg) {
-  return arg instanceof Buffer;
+  return Buffer.isBuffer(arg);
 }
 exports.isBuffer = isBuffer;
 
